### PR TITLE
Update to Querydsl 4

### PR DESCRIPTION
**Pull request by timowest**

### CODE REVIEW

File: querydsl-elasticsearch/src/test/java/com/querydsl/elasticsearch/ElasticsearchQueryTest.java [hunk 1]
```diff
@@ -132,14 +131,14 @@ public void Starts_With_and_Between() {
 
     @Test
     public void Exists() {
-        assertTrue(where(user.firstName.eq("Jaakko")).exists());
-        assertFalse(where(user.firstName.eq("JaakkoX")).exists());
-        assertTrue(where(user.id.eq(u1.getId())).exists());
+        assertTrue(where(user.firstName.eq("Jaakko")).fetchCount() > 0);
+        assertFalse(where(user.firstName.eq("JaakkoX")).fetchCount() == 0);
```
Comment (johnktims) [inline querydsl-elasticsearch/src/test/java/com/querydsl/elasticsearch/ElasticsearchQueryTest.java]: This should be `assertTrue`.

**Comment by johnktims** [comment]
There's still trailing whitespace in the apt/assembly.xml files and imports definitely need to be organized but that can be done in another PR.

How come these modules are kept outside of querydsl proper? I'm just trying to understand which contributions are allowed into querydsl and which ones end up in contrib.

**Comment by timowest** [comment]
querydsl-contrib contains modules that we don't officially support. They are released in sync with the minor releases. With Querydsl 4 that didn't happen yet, but will be done.

Contributions to the main modules go into `querydsl` and most new integrations into `querydsl-contrib`.

**Comment by johnktims** [comment]
Don't we need to maintain multiple branches so that we can keep releasing the contrib modules for the 3.x line?

**Comment by timowest** [comment]
That's true, we could introduce a `QUERYDSL_3` branch here as well.

**Comment by kevinleturc** [comment]
Hello,

Do you need some help to finish this ?
Is it possible to release two versions of querydsl-contrib ? For 3.x and 4.x versions ?

Regards

**Comment by johnktims** [comment]
@kevinleturc I just pushed a QUERYDSL_3 branch so we can make separate releases for 3.x and 4.x.

@Shredder121, feel free to merge this PR.

**Comment by Shredder121** [comment]
Looks good, I'll merge this.

**Pull request merged**